dev/ToastNotifications/NotificationProperties.cpp

```diff
@@ -0,0 +1,148 @@
+    CopyMemory(m_payload.data(), payloadAsSimpleString.c_str(), payloadAsSimpleString.size());
+
+    m_notificationId = toastNotification.ToastId();
+
+    m_tag = toastNotification.Tag();
+    m_group = toastNotification.Group();
+
+    auto expiry = toastNotification.ExpirationTime();
+    m_expiry = winrt::clock::to_file_time(expiry);
+
+    auto arrivalTime = winrt::clock::now();
+    m_arrivalTime = winrt::clock::to_file_time(arrivalTime);
+
+    m_expiresOnReboot = toastNotification.ExpiresOnReboot();
+
+    // TODO: Uncomment once ToastProgressData is implemented
+    // m_progressData = toastNotification.ProgressData();
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_NotificationId(_Out_ unsigned int* notificationId) noexcept
+{
+    auto lock = m_lock.lock_shared();
+
+    *notificationId = m_notificationId;
+    return S_OK;
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_Payload(_Out_ unsigned int* payloadSize, _Out_ byte** payload) noexcept try
+{
+    auto lock = m_lock.lock_shared();
+
+    size_t tempPayloadSize = m_payload.size();
+
+    auto tempPayload = wil::unique_cotaskmem_array_ptr<byte>(static_cast<byte*>(CoTaskMemAlloc(tempPayloadSize)), tempPayloadSize);
+    CopyMemory(tempPayload.data(), m_payload.get(), tempPayloadSize);
+
+    *payloadSize = static_cast<unsigned int>(tempPayloadSize);
+    *payload = tempPayload.release();
+    return S_OK;
+}
+CATCH_RETURN()
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_PayloadSize(_Out_ unsigned int* payloadSize) noexcept
+{
+    auto lock = m_lock.lock_shared();
+
+    *payloadSize = static_cast<unsigned int>(m_payload.size());
+    return S_OK;
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_Tag(_Out_ HSTRING* tag) noexcept try
+{
+    auto lock = m_lock.lock_shared();
+
+    *tag = safe_make_unique_hstring(m_tag.c_str()).release();
+    return S_OK;
+}
+CATCH_RETURN()
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_Group(_Out_ HSTRING* group) noexcept try
+{
+    auto lock = m_lock.lock_shared();
+
+    *group = safe_make_unique_hstring(m_group.c_str()).release();
+    return S_OK;
+}
+CATCH_RETURN()
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_Expiry(_Out_ unsigned long long* expiry) noexcept
+{
+    auto lock = m_lock.lock_shared();
+
+    *expiry = static_cast<ULONGLONG>(m_expiry.dwHighDateTime) << 32 | static_cast<ULONGLONG>(m_expiry.dwLowDateTime);
+    return S_OK;
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_ArrivalTime(_Out_ unsigned long long* arrivalTime) noexcept
+{
+    auto lock = m_lock.lock_shared();
+
+    *arrivalTime = static_cast<ULONGLONG>(m_arrivalTime.dwHighDateTime) << 32 | static_cast<ULONGLONG>(m_arrivalTime.dwLowDateTime);
+    return S_OK;
+}
+
+STDMETHODIMP_(HRESULT __stdcall) NotificationProperties::get_BootId(_Out_ unsigned long long* bootId) noexcept
+{
+    *bootId = 0;
```

Comment:
Please declare a variable called m_bootId. Set that in the ctor. It's equal to 0 still but thats okay.

Reply:
What are the benefits of doing this way or what are why trying to achieve?

Reply:
When we have a public API available to generate bootId in Reunion SDK, you can use that at the top of the stack without plumbing a bunch of changes here. It makes fixes less risky and more manageable.

Reply:
The same idea for CV too

Reply:
Fair enough for bootId.

For CV, I would imagine it will have its own datatype when introduced. Then, changes will be done anyway at the getter function to extract the CV as hstring. So, for now I'll leave it as is.